### PR TITLE
[platform] Add support for Dell Z9100

### PR DESCRIPTION
**Pull request by nirenjan**

### CODE REVIEW

File: device/dell/x86_64-dell_z9100_c2538-r0/plugins/eeprom.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# Dell Z9100
+#
+# Platform and model specific eeprom subclass, inherits from the base class,
+# and provides the followings:
+# - the eeprom format definition
+# - specific encoder/decoder if there is special need
+#############################################################################
+
+try:
+    import eeprom_tlvinfo
```
Comment (lguohan) [inline device/dell/x86_64-dell_z9100_c2538-r0/plugins/eeprom.py]: can you check #342 ? 

from sonic_eeprom import eeprom_tlvinfo

File: device/dell/x86_64-dell_z9100_c2538-r0/plugins/sfputil.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    """Platform specific sfputil class"""
+
+    port_start = 0
+    port_end = 32
+    ports_in_block = 32
+
+    port_to_eeprom_mapping = {}
+    port_to_i2c_mapping = {
+        0: 18,
+        1: 18,
```
Comment (lguohan) [inline device/dell/x86_64-dell_z9100_c2538-r0/plugins/sfputil.py]: there seems to duplication here.

File: device/dell/x86_64-dell_z9100_c2538-r0/plugins/sfputil.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    from sfputilbase import sfputilbase
+except ImportError, e:
+    raise ImportError (str(e) + "- required module not found")
+
+
+class sfputil(sfputilbase):
+    """Platform specific sfputil class"""
+
+    port_start = 0
+    port_end = 32
```
Comment (lguohan) [inline device/dell/x86_64-dell_z9100_c2538-r0/plugins/sfputil.py]: should be 31?

File: device/dell/x86_64-dell_z9100_c2538-r0/plugins/eeprom.py (hunk 1 above)
Comment (nirenjan) [inline device/dell/x86_64-dell_z9100_c2538-r0/plugins/eeprom.py]: My mistake. I didn't see #342 before I raised this... I can decline this in favor of that one.

**Comment by nirenjan** [comment]
I'll test this again, but this was written before #316, so your changes to #342 probably are correct.

**Comment by nirenjan** [comment]
Declining in favor of #342

**Pull request closed without merging**